### PR TITLE
add fee estimate

### PR DESCRIPTION
**Pull request by marshacb**
* adds method to calculate fee estimate for a given extrinsic

closes: [issue #32](https://github.com/paritytech/asset-transfer-api/issues/32)

### CODE REVIEW

File: src/AssetsTransferApi.ts [hunk 1]
```diff
@@ -156,6 +160,44 @@ export class AssetsTransferApi {
 			opts?.format
 		);
 	}
+	/**
+	 * Fetch estimated fee information for an extrinsic
+	 *
+	 * @param tx TxResult<T>
```
Comment (TarikGul) [inline src/AssetsTransferApi.ts:166]: `@param format` as well.

File: src/AssetsTransferApi.ts [hunk 2]
```diff
@@ -156,6 +160,44 @@ export class AssetsTransferApi {
...
+	 */
+	public async fetchFeeInfo<T extends Format>(
+		tx: SubmittableExtrinsic<'promise', ISubmittableResult>,
+		format?: T
+	): Promise<RuntimeDispatchInfo | RuntimeDispatchInfoV1 | null> {
+		const { _api: api } = this;
```
Comment (TarikGul) [inline src/AssetsTransferApi.ts:172]: Should we keep this in line with the rest of the API where its `_api` to denote its read only, if you think its easier to read like `api` then I am happy to have a follow up PR change the rest of the API.

**Review by TarikGul — APPROVED** [review]
lgtm, great work :)

File: src/AssetsTransferApi.ts (hunk 2 above)
Comment (marshacb) [reply src/AssetsTransferApi.ts:172]: totally. Ill change this to `_api`

**Pull request merged**